lms/services/application_instance.py

```diff
@@ -31,6 +31,27 @@ def get(self, consumer_key=None) -> ApplicationInstance:
 
         return application_instance
 
+    def create(  # pylint:disable=too-many-arguments
+        self, lms_url, email, developer_key, developer_secret, aes_secret
+    ):
+        # If either one of developer_key or developer_secret is missing, then we
+        # don't save the other one either.
+        if not developer_key or not developer_secret:
+            developer_key = None
+            developer_secret = None
+
+        ai = ApplicationInstance.build_from_lms_url(
```

Comment:
This could be on the service or stay on the model. I left it on the model which I think it makes sense.